From: Alistair Popple <apopple@nvidia.com>
Subject: mm/mlock: skip ZONE_DEVICE PMDs during mlock
Date: Tue, 7 Jan 2025 14:42:36 +1100

At present mlock skips ptes mapping ZONE_DEVICE pages.  A future change to
remove pmd_devmap will allow pmd_trans_huge_lock() to return ZONE_DEVICE
folios so make sure we continue to skip those.

Link: https://lkml.kernel.org/r/735ecdeef28a03b0e9114b2435e0ef7af7a73c66.1736221254.git-series.apopple@nvidia.com
Signed-off-by: Alistair Popple <apopple@nvidia.com>
Acked-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mlock.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/mlock.c~mm-mlock-skip-zone_device-pmds-during-mlock
+++ a/mm/mlock.c
@@ -368,6 +368,8 @@ static int mlock_pte_range(pmd_t *pmd, u
 		if (is_huge_zero_pmd(*pmd))
 			goto out;
 		folio = pmd_folio(*pmd);
+		if (folio_is_zone_device(folio))
+			goto out;
 		if (vma->vm_flags & VM_LOCKED)
 			mlock_folio(folio);
 		else
_
